js/client-rooms.js

```diff
@@ -80,20 +80,24 @@
 				app.roomsFirstOpen = 2;
 			}
 		},
+
 		renderRoomBtn: function (roomData) {
 			var id = toID(roomData.title);
-			var buf = '<div><a href="' + app.root + id + '" class="ilink"><small style="float:right">(' + Number(roomData.userCount) + ' users)</small><strong><i class="fa fa-comment-o"></i> ' + BattleLog.escapeHTML(roomData.title) + '<br /></strong><small>' + BattleLog.escapeHTML(roomData.desc || '');
+			if (!(roomData.subRooms && roomData.subRooms.length))
+				var buf = '<div class="ilink" style="margin-bottom:5px"><a style="color:inherit" href="' + app.root + id + '"><small style="float:right">(' + Number(roomData.userCount) + ' users)</small><strong><i class="fa fa-comment-o"></i> ' + BattleLog.escapeHTML(roomData.title) + '<br /></strong><small>' + BattleLog.escapeHTML(roomData.desc || '') + '</a></div>';
 			if (roomData.subRooms && roomData.subRooms.length) {
-				buf += '<br/><i class="fa fa-level-up fa-rotate-90"></i> Subrooms: <strong>';
+				buf = '<div class="ilink" style="margin-bottom:-2px"><a style="color:inherit" href="' + app.root + id + '"><small style="float:right">(' + Number(roomData.userCount) + ' users)</small><strong><i class="fa fa-comment-o"></i> ' + BattleLog.escapeHTML(roomData.title) + '<br /></strong><small>' + BattleLog.escapeHTML(roomData.desc || '') + '<br><i>Subrooms: </i></a><strong></div>';
+				buf += '<span style="padding-right:20px"></span><i class="arrowsubrooms" style="padding-right:5px"><i class="fa fa-level-up fa-rotate-90"></i></i>';
 				for (var i = 0; i < roomData.subRooms.length; i++) {
-					if (i) buf += ', ';
-					buf += '<i class="fa fa-comment-o"></i> ' + BattleLog.escapeHTML(roomData.subRooms[i]);
+					if (i) buf += '';
+					buf += ' <a class="subroom" style="font-size:11px;display:inline" href="/' + roomData.subRooms[i].split(" ").join("").toLowerCase() + '"><div class="subroom" style="min-width:25%;position:relative;display:inline-block;padding:2px 4px 2px 4px;margin:2px 0px 5px 0px;border-top-left-radius:0px;border-top-right-radius:0px;border-top:none"><i class="fa fa-comment-o" ></i> ' + BattleLog.escapeHTML(roomData.subRooms[i]) + '</div></a>';
```

Comment:
This seems like a lot of nested HTML. Is this necessary?

Reply:
No, I cleared that up